packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -785,7 +785,10 @@ export class CallContext {
     this.modifyState((draft: CallClientState) => {
       const call = draft.calls[this._callIdHistory.latestCallId(callId)];
       if (call) {
-        this.processNewCaption(call.captionsFeature.captions, convertFromSDKToCaptionInfoState(caption));
+        const currentCaptionLanguage = call.captionsFeature.currentCaptionLanguage
+        if (caption.captionLanguage.toUpperCase() === currentCaptionLanguage.toUpperCase() || currentCaptionLanguage === '' || currentCaptionLanguage === undefined){
```

Comment:
when is a caption received different to the selected caption language?

Reply:
When caption language is changed, we receive 2 captions, one translated and one not translated, so we will need to filter out the non translated one by checking caption language

And toUpperCase() is because for chinese, we receive 'zh-hans' from caption and 'zh-Hans' from captionsFeature